json-rpc/src/tests/unit_tests.rs

```diff
@@ -1193,6 +1237,58 @@ fn test_get_transactions() {
     }
 }
 
+#[test]
+fn test_get_transactions_with_proof() {
+    let (mock_db, client, mut runtime) = create_database_client_and_runtime();
+
+    let version = mock_db.get_latest_version().unwrap();
+    let page = 800usize;
+
+    for base_version in (0..version)
+        .map(u64::from)
+        .take(page)
+        .collect::<Vec<_>>()
+        .into_iter()
+    {
+        println!(
+            "Working on base_version {} out of {}",
+            base_version, version
+        );
+        let mut batch = JsonRpcBatch::default();
+        batch.add_get_transactions_with_proofs_request(base_version, page as u64);
+        let result = execute_batch_and_get_first_response(&client, &mut runtime, batch);
+        println!("No panic {}", base_version);
+
+        let received_proof = TransactionsWithProofsView::from_response(result).unwrap();
+
+        // if it's empty then no need to process it.
+        if received_proof.serialized_transactions.is_empty() {
```

Comment:
this is not a good way to do test, what if all of responses' serialized_transactions is empty?

I think we can remove this test, add one test to json-rpc/tests/integration_test instead.
and the test should validate the proof works with respond transaction blobs